@here/harp-features-datasource/lib/FeaturesDataSource.ts

```diff
@@ -38,6 +43,35 @@ export class FeaturesDataSource extends OmvDataSource {
         this.addTileBackground = false;
     }
 
+    /**
+     * This method allows to directly add a GeoJSON wihout using [[MapViewFeature]] instances. It
+     * also overwrites existing features in this data source. To add a GeoJSON without overwriting
+     * the data source, one should loop through it to create [[MapViewFeature]] and add them with
+     * the `add` method.
+     *
+     * @param geojson A javascript object matching the GeoJSON specification.
+     */
+    setFromGeojson(geojson: FeatureCollection | GeometryCollection | Feature) {
+        if (geojson.type === "FeatureCollection") {
+            this.m_featureCollection = geojson;
+        } else if (geojson.type === "Feature") {
+            this.m_featureCollection = DEFAULT_GEOJSON;
```

Comment:
You're adding objects to `DEFAULT_GEOJSON` instance.

~You probably wanted
```
{ ...DEFAULT_GEOJSON }
```
But this will always write to same instance of `DEFAULT_GEOJSON.features` ...
So, this `DEFAULT_GEOJSON` doesn't work here at all, you just to call `clear`.